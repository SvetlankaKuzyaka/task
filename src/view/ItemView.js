export default class ItemView {
  constructor(itemData) {
    this.itemData = itemData;
  }

  renderItem() {
    const subContainer = document.querySelector('.container');
    const containerTwo = document.createElement('div');
    containerTwo.classList.add('item_container');
    const markup = `
      <div class = "item">
        <div class="title">
          <a href="https://www.youtube.com/watch?v=${this.itemData.videoId}" target="_blank">${this.itemData.title}</a>
        </div>
        <img src="${this.itemData.url}">
        <div class="info">
          <ul>
            <li class="author"><span>channel:</span> ${this.itemData.channelTitle}</li>
            <li class="data_item"><span>data:</span> ${this.itemData.publishedAt.substr(0, 10)}</li>
            <li class="quantity_view"><span>view:</span> ${this.itemData.statistics}</li>
          </ul>
        </div>
        <div class="description">
          <span class="description">${this.itemData.description}</span>
        </div>
      </div>
    `;
    containerTwo.innerHTML = markup;
    subContainer.appendChild(containerTwo);
  }
}
