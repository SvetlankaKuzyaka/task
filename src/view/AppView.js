import ItemView from './ItemView';

export default class AppView {
  static renderInitialPage() {
    const upContainer = document.createElement('div');
    upContainer.classList.add('upcontainer');
    const searchField = document.createElement('input');
    const buttonSearch = document.createElement('button');
    const mainContainer = document.createElement('div');
    mainContainer.classList.add('main_container');
    searchField.setAttribute('name', 'search');
    buttonSearch.setAttribute('name', 'send');
    buttonSearch.textContent = 'SEND';
    upContainer.appendChild(searchField);
    upContainer.appendChild(buttonSearch);
    document.body.appendChild(upContainer);
    document.body.appendChild(mainContainer);
    const container = document.createElement('div');
    container.classList.add('container');
    mainContainer.appendChild(container);
    AppView.M = 4;
    AppView.count = 0;
    const subContainer = document.createElement('div');
    subContainer.classList.add('subcontainer');
    subContainer.innerHTML = `
      <button class="prev"><</button>
      <span class="page"></span>
      <button class="next">></button>`;
    document.body.appendChild(subContainer);
    document.querySelector('.page').textContent = AppView.count;
  }

  static clear() {
    const container = document.querySelector('.container');
    AppView.count = 0;
    container.style.setProperty('--i', AppView.count);
    document.querySelector('.page').textContent = AppView.count;
    document.querySelector('.next').setAttribute('title', AppView.count + 1);
    document.querySelector('.prev').setAttribute('title', AppView.count > 0 ? AppView.count - 1 : 'нет');
    if (AppView.count !== 0) {
      const arr = Array.prototype.slice.apply(container.children);
      arr.forEach(element => container.removeChild(element));
    }
    container.innerHTML = '';
  }

  static renderData(data) {
    data.forEach((element) => {
      new ItemView(element).renderItem();
    });
  }
}
