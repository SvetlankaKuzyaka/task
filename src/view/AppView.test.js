import AppView from './AppView';
import ItemView from './ItemView';

describe('ItemView constructor', () => {
  it('Should create new ItemView', () => {
    const itemData = {
      channelTitle: 'JS Films',
      description: 'Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH ***Buy my MERCHANDISE***: https://bit.ly/2DedHWW OFFICIAL WEBSITE: ...',
      publishedAt: '2019-05-22T03:30:00.000Z',
      statistics: 123456,
      title: 'Is India ready for electric vehicles?',
      url: 'https://i.ytimg.com/vi/hzkKe94CkJI/hqdefault.jpg',
      videoId: 'hsRPx3-00xo',
    };
    const object = {
      itemData: {
        videoId: 'hsRPx3-00xo',
        title: 'Is India ready for electric vehicles?',
        publishedAt: '2019-05-22T03:30:00.000Z',
        channelTitle: 'JS Films',
        description: 'Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH ***Buy my MERCHANDISE***: https://bit.ly/2DedHWW OFFICIAL WEBSITE: ...',
        url: 'https://i.ytimg.com/vi/hzkKe94CkJI/hqdefault.jpg',
        statistics: 123456,
      },
    };
    const result = new ItemView(itemData);
    expect(result).toEqual(object);
  });
});

describe('AppView.renderInitialPage', () => {
  it('Should be a Function', () => {
    expect(AppView.renderInitialPage).toBeInstanceOf(Function);
  });

  it('Should set init right parametres', () => {
    AppView.renderInitialPage();
    expect(AppView.count).toEqual(0);
    expect(AppView.M).toEqual(4);
  });
});

describe('AppView.clear', () => {
  it('Should be a Function', () => {
    expect(AppView.clear).toBeInstanceOf(Function);
  });

  it('Should set init right parametres', () => {
    AppView.clear();
    expect(AppView.count).toEqual(0);
  });

  it('Should clear data correctly', () => {
    AppView.clear();
    expect(document.querySelector('.container')).toMatchSnapshot();
  });
});

describe('AppView.renderData', () => {
  it('Should be a Function', () => {
    expect(AppView.renderData).toBeInstanceOf(Function);
  });
});

describe('ItemView.prototype.renderItem', () => {
  it('Should be a Function', () => {
    expect(ItemView.prototype.renderItem).toBeInstanceOf(Function);
  });

  it('Should render data correctly', () => {
    const object = {
      itemData: {
        videoId: 'hsRPx3-00xo',
        title: 'Is India ready for electric vehicles?',
        publishedAt: '2019-05-22T03:30:00.000Z',
        channelTitle: 'JS Films',
        description: 'Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH ***Buy my MERCHANDISE***: https://bit.ly/2DedHWW OFFICIAL WEBSITE: ...',
        url: 'https://i.ytimg.com/vi/hzkKe94CkJI/hqdefault.jpg',
        statistics: 123456,
      },
    };
    ItemView.prototype.renderItem.call(object);
    expect(document.querySelector('.item_container')).toMatchSnapshot();
  });
});
