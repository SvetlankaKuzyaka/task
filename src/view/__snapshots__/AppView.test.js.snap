// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.clear Should clear data correctly 1`] = `
<div
  class="container"
/>
`;

exports[`ItemView.prototype.renderItem Should render data correctly 1`] = `
<div
  class="item_container"
>
  
      
  <div
    class="item"
  >
    
        
    <div
      class="title"
    >
      
          
      <a
        href="https://www.youtube.com/watch?v=hsRPx3-00xo"
        target="_blank"
      >
        Is India ready for electric vehicles?
      </a>
      
        
    </div>
    
        
    <img
      src="https://i.ytimg.com/vi/hzkKe94CkJI/hqdefault.jpg"
    />
    
        
    <div
      class="info"
    >
      
          
      <ul>
        
            
        <li
          class="author"
        >
          <span>
            channel:
          </span>
           JS Films
        </li>
        
            
        <li
          class="data_item"
        >
          <span>
            data:
          </span>
           2019-05-22
        </li>
        
            
        <li
          class="quantity_view"
        >
          <span>
            view:
          </span>
           123456
        </li>
        
          
      </ul>
      
        
    </div>
    
        
    <div
      class="description"
    >
      
          
      <span
        class="description"
      >
        Get my Sunglasses from EyewearLabs: https://goo.gl/8oZwgH ***Buy my MERCHANDISE***: https://bit.ly/2DedHWW OFFICIAL WEBSITE: ...
      </span>
      
        
    </div>
    
      
  </div>
  
    
</div>
`;
